src/java/htsjdk/samtools/util/SimpleInterval.java

```diff
@@ -0,0 +1,45 @@
+package htsjdk.samtools.util;
+
+
+/**
+ * minimal immutable class representing a 1-based closed ended genomic interval
+ */
+public class SimpleInterval implements Locatable {
+
+    private final int start;
+    private final int end;
+    private final String contig;
+
+    public SimpleInterval(final String contig, final int start, final int end){
+        if(contig == null){
+            throw new IllegalArgumentException("contig cannot be null");
```

Comment:
I know it sounds like an oxymoron, but we actually have this concept in the GATK. Eg., if the user passes in `-L unmapped`, the GATK will create an interval object (a `GenomeLoc`, in our case) to represent this "unmapped" interval, and the engine, when given this unmapped interval object, will then set up a traversal over the unmapped reads in the input file(s). 

If a `SamRecord` can be unmapped, yet implements `Locatable`, doesn't that imply that "Locatables" in general need some way of indicating the absence of a location (ie., something like an `isUnmapped()` method)?